pkg/store/bucket.go

```diff
@@ -2869,12 +2878,13 @@ func (r *bucketIndexReader) fetchExpandedPostingsFromCache(ctx context.Context,
 	if !hit {
 		return false, nil, nil
 	}
-	if err := bytesLimiter.Reserve(uint64(len(dataFromCache))); err != nil {
-		return false, nil, httpgrpc.Errorf(int(codes.ResourceExhausted), "exceeded bytes limit while loading expanded postings from index cache: %s", err)
+	if bytesLimiter != nil {
```

Comment:
Let's remove the nil check here. It is probably redundant after we changing back to use one limiter only

Reply:
Are we confident that bytesLimiter won't be null, as Series/LabelNames/LabelValues functions all create bytes limiter from the factory and pass down? Then I think we should check at least if the factory isn't null, or assign `NewBytesLimiterFactory(0)` if null is passed (and probably similar for Chunks and Series limiter factories). What do you think?

Reply:
Since Thanos is only used itself and by downstream projects like Cortex, those project should be responsible to make sure to not return a nil limiter. At least return a noop limiter. Maybe we should add a comment about the contract.

> Then I think we should check at least if the factory isn't null, or assign NewBytesLimiterFactory(0) if null is passed (and probably similar for Chunks and Series limiter factories). What do you think?

But it can be done at the caller side when initializing the bucket store? At least I don't see why we need to do it in Thanos itself since Thanos ensures that the factory is not nil https://github.com/thanos-io/thanos/blob/main/cmd/thanos/store.go#L436.

If it is possible to have nil factory in Cortex we can do it on Cortex. But for me, it should be safe to skip.

Reply:
ok, sounds good to me